client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { get, merge } from 'lodash';
+
+import { decorrelatedJitter as defaultDelay } from './delays';
+import { default as defaultPolicy } from './policies';
+
+const isGetRequest = request => 'GET' === get( request, 'method', '' ).toUpperCase();
+
+export const retryOnFailure = ( getDelay = defaultDelay ) => inboundData => {
+	const { nextError, originalRequest, store: { dispatch } } = inboundData;
+
+	// if the request came back successfully
+	// then we have no need to intercept it
+	// flush our records of it
+	if ( ! nextError ) {
+		return inboundData;
+	}
+
+	// otherwise check if we should try again
+	if ( ! isGetRequest( originalRequest ) ) {
```

Comment:
what's considered an error? I don't think we want to retry 4xx level errors, they're valid responses, not network errors. I think we only want to retry 5xx level errors and timeouts.